src/main/java/linkedList/LinkedList.java

```diff
@@ -0,0 +1,93 @@
+package linkedList;
+
+public class LinkedList<T> {
+    Node head;
+    Node tail;
+    int currentIndex = 0;
+
+    public void insert(T data) {
+        Node node = new Node();
+        node.data = data;
+        node.next = null;
+        node.prev = null;
+        if (head == null) {
+            head = node;
+            tail = node;
+        } else {
+            Node n = head;
+            while (n.next != null) {
```

Comment:
Since there's a pointer to `tail`, there's no need to iterate through the whole list to add an element in the end. 
One can just set a Node to `tail.next` and move `tail = tail.next` and that's it